src/main/java/testing/IntArrayOperationsV2.java

```diff
@@ -0,0 +1,10 @@
+package testing;
+
+public class IntArrayOperationsV2 {
+    public static boolean isSorted(int[] values) {
```

Comment:
I think in this case the algorithm is also a lot simpler (though it's admittedly non-composable)